data/graphicsmagick/test.sh

```diff
@@ -0,0 +1,491 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: GraphicMagick testsuite
+# Maintainer: Ivan Lausuch <ilausuch@suse.com>
+#
+# Usage; test.sh path_of_resources
+
+#####################
+# Download resorces
+#####################
+
+resources=(
+  black.png
+  blue.png
+  check_compare.pl
+  check_compare_PSNR.pl
+  check_size.pl
+  degradation_1bit.png
+  degradation_grayscale.png
+  degradation_monochrome.png
+  degradation.png
+  logo-primary.png
+  magenta.png
+  montage1.png
+  montage2.png
+  noise_blur_10.png
+  noise_emboss_10.png
+  noise_with_tile_logo.png
+  noise.gif
+  quadrants500x500_transparent_blue.png
+  quadrants500x500.png
+  quadrants500x500.xcf
+  red_circle.png
+  red.png
+  script.html
+  suse_tail_composite.png
+  suse_tail_contrast_40.png
+  suse_tail_contrast_90.png
+  suse_tail_gamma_0.png
+  suse_tail_sharpen5.png
+  suse_tail.jpg
+  test_animation.gif
+  test_text_variables.png
+  test_text.png
+  test_texture.png
+  test_virtual_pixel_constant.png
+  test_virtual_pixel_edge.png
+  test_virtual_pixel_tile.png
+  test_virtual_pixel_mirror.png
+  white.png
+)
+
+for item in ${resources[*]}
+do
+  wget --quiet "$1/$item" -O $item
+done
+
+##########################
+# Prepare some resources
+#########################
+
+gm convert degradation.png degradation.jpg
+gm convert noise.gif noise.jpg
+gm convert noise.gif -resize 500x500 noise500x500.png
+gm convert black.png -resize 10x10 black10x10.png
+gm convert -crop 500x250+0+0! quadrants500x500.png quadrants_up_500x250.png
+
+gm convert red.png frame1.gif
+gm convert black.png frame2.gif
+gm convert blue.png frame3.gif
+gm convert white.png frame4.gif
+
+##########################
+# Helpers
+##########################
+
+function compare {
+  metric=$1
+  image1=$2
+  image2=$3
+  tolerance=$4
+  inversion=$5
+  channel=$6
+
+  if [ "$metric" = "PSNR" ]; then
+    script="check_compare_PSNR.pl"
+  else
+    script="check_compare.pl"
+  fi
+
+  if [ -z "$channel" ]; then
+    channel="Total"
+  fi
+
+  if [ ! -f $image1 ]; then
+    echo "KO"
+    return
+  fi
+
+  if [ ! -f $image2 ]; then
+    echo "KO"
+    return
+  fi
+
+  gm compare -metric $metric $image1 $image2 | grep $channel | perl $script $tolerance $inversion
+}
+
+function convert_and_compare {
+  image1=$1
+  image2=$2
+  tolerance=$3
+  options=$4
+  image3=$5
+
+  if [ "$options" = '-' ]; then
+    options=""
+  fi
+
+  if [ -z "$image3" ]; then
+    image3=$image2
+  fi
+
+  options=$(echo $options | perl -pe "s/(_)/ /g")
+  options=$(echo $options | perl -pe "s/(\^)/%/g")
```

Comment:
```sh
options="${options//_/ }"
options="${options//^/%}"
```
if the script is called with bash

Reply:
Thanks Tina for your comments. WIP

Reply:
just checked, it is not POSIX compliant to use string replacements :-/

Reply:
Done, thanks
Added #!/bin/bash

Reply:
#!/bin/bash added
I understand with this is enough. The test is in green.